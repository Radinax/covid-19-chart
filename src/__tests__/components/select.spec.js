import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import '@testing-library/jest-dom'
import Select from '../../components/select'

const name = 'Select view'
const values = ['Venezuela Map', 'Chart by states in Venezuela', 'Chart by Countries', 'World Map']
const value = values[0]
const onChange = () => null

describe('if Select is working correctly', () => {
  it('renders', () => {
    const { asFragment } = render(
      <Select name={name} values={values} value={value} onChange={onChange}/>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('that the value shown is "Venezuela Map"', () => {
    const { getByTestId } = render(
      <Select name={name} values={values} value={value} onChange={onChange}/>
    )
    expect(getByTestId('select').value).toBe(value)
  })

  test('when user changes the option when it clicks the element', () => {
    const handleChange = e => expect(e.target.value).toBe(values[2])

    const { asFragment, container } = render(
      <Select name={name} values={values} value={value} onChange={handleChange}/>
    )
    const select = container.querySelector(`select[name="${name}"]`)
    wait(() => {
      fireEvent.change(select, {
        target: {
          value: values[2]
        }
      })
    })
    expect(asFragment()).toMatchSnapshot()
  })
})