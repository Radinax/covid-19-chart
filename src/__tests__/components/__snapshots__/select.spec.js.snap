// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if Select is working correctly renders 1`] = `
<DocumentFragment>
  <select
    data-testid="select"
    name="Select view"
    required=""
    style="height: 1rem; margin: 1rem;"
  >
    <option
      value="Venezuela Map"
    >
      Venezuela Map
    </option>
    <option
      value="Chart by states in Venezuela"
    >
      Chart by states in Venezuela
    </option>
    <option
      value="Chart by Countries"
    >
      Chart by Countries
    </option>
    <option
      value="World Map"
    >
      World Map
    </option>
  </select>
</DocumentFragment>
`;

exports[`if Select is working correctly when user changes the option when it clicks the element 1`] = `
<DocumentFragment>
  <select
    data-testid="select"
    name="Select view"
    required=""
    style="height: 1rem; margin: 1rem;"
  >
    <option
      value="Venezuela Map"
    >
      Venezuela Map
    </option>
    <option
      value="Chart by states in Venezuela"
    >
      Chart by states in Venezuela
    </option>
    <option
      value="Chart by Countries"
    >
      Chart by Countries
    </option>
    <option
      value="World Map"
    >
      World Map
    </option>
  </select>
</DocumentFragment>
`;
